assets/js/planning-update.js

```diff
@@ -20,6 +20,32 @@ function selectTableBox($tableBox) {
   colorTableBox($tableBox);
 }
 
+function handleSlotBoxClick(event) {
+  event.stopImmediatePropagation();
+
+  const $slotBox = event.target === 'input' ? $(this).closest('.slot-box') : $(this);
+  selectTableBox($slotBox);
+
+  const $planning = $('.planning');
+  const $lastClickedTd = $planning.find('.prev-clicked');
+
+  if ($lastClickedTd && event.shiftKey) {
+    handleShiftClick($planning, $slotBox, $lastClickedTd);
+    $lastClickedTd.removeClass('prev-clicked');
+    $lastClickedTd.removeClass('highlight');
+  }
+
+  $slotBox.addClass('prev-clicked');
```

Comment:
you must remove old `prev-clicked` even if the shift key is not currently pressed